Index: android/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/android/src/main/AndroidManifest.xml b/android/src/main/AndroidManifest.xml
--- a/android/src/main/AndroidManifest.xml	
+++ b/android/src/main/AndroidManifest.xml	(date 1667663205039)
@@ -2,6 +2,4 @@
           package="com.buddybosscustomcode">
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
     <uses-permission android:name="android.permission.ACCESS_BACKGROUND_SERVICE"/>
-<!--    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />-->
 </manifest>
-
Index: Screens/HomeContent.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Screens/HomeContent.js b/Screens/HomeContent.js
--- a/Screens/HomeContent.js	
+++ b/Screens/HomeContent.js	(date 1667608768644)
@@ -36,7 +36,7 @@
     const achievementReducer = useSelector((state) => state.onenergyReducer?state.onenergyReducer.achievementReducer:null);
     const postReducer = useSelector((state) => state.postReducer?state.postReducer:null);
     const dispatch = useDispatch();
-    console.log(achievementReducer)
+
     const onFocusHandler=() =>
     {
         try
@@ -53,6 +53,8 @@
     const _handleAppStateChange = async () => {
         if(user) {
             if((Platform.OS === "android" && AppState.currentState==='background') || (Platform.OS === "ios" && AppState.currentState==='inactive')) {
+
+                console.log('updateRemote')
                 if (progressReducer.latestUpdate && progressReducer.lastUpload && progressReducer.latestUpdate > progressReducer.lastUpload || !progressReducer.lastUpload) {
                     let achievements = {
                         'achievements': [],
Index: Screens/QuotesScreen.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Screens/QuotesScreen.js b/Screens/QuotesScreen.js
--- a/Screens/QuotesScreen.js	
+++ b/Screens/QuotesScreen.js	(date 1667582388603)
@@ -9,7 +9,8 @@
     ActivityIndicator,
     PermissionsAndroid,
     SafeAreaView,
-    FlatList
+    FlatList,
+    Share
 } from "react-native";
 import {connect, useSelector, useDispatch} from "react-redux";
 import {getApi} from "@src/services";
@@ -19,6 +20,7 @@
 import ScalableImage from "../Components/ScalableImage";
 import RNFetchBlob from 'rn-fetch-blob';
 import analytics from '@react-native-firebase/analytics';
+import TouchableScale from "../Components/TouchableScale";
 
 const QuotesScreen = props => {
     const [loading, setLoading] = useState(false);
@@ -52,6 +54,7 @@
                         title: item.title,
                         format: item.format,
                         excerpt: item.excerpt,
+                        share_url: item.meta_box.share_url,
                         author: item._embedded['author'][0].name,
                         avatar: item._embedded['author'][0].avatar_urls['24'],
                         image: item._embedded['wp:featuredmedia'][0].source_url,
@@ -175,6 +178,26 @@
     const renderItem = ({ item }) => (
         <View style={styles.container}>
             <ScalableImage width={windowWidth} source={{uri: item.image?item.image:''}} />
+            {item.share_url?
+            <TouchableScale
+                onPress={() => {
+                    Share.share({
+                        url: item.share_url
+                    }, {})
+                }}
+            >
+                <View style={[styles.buttonShare, styles.boxShadow]}>
+                    <Text
+                        style={{ fontSize: scale(20), color: '#FFF', textShadowColor: 'grey', textShadowRadius: 1, textShadowOffset: {
+                                width: -1,
+                                height: 1
+                            }}}
+                    >
+                        SHARE
+                    </Text>
+                </View>
+            </TouchableScale>
+                :null}
         </View>
     );
     return (
@@ -233,10 +256,20 @@
         justifyContent: 'flex-start',
         backgroundColor: '#f6f6f8',
     },
+    buttonShare: {
+        marginVertical: 15,
+        borderRadius: 9,
+        alignItems: 'center',
+        justifyContent: 'center',
+        width: scale(200),
+        height: scale(50),
+        backgroundColor: '#8c78ff',
+    },
     view: {
         padding: 0,
         flex: 1,
         justifyContent:"flex-start",
+        alignItems: "center"
     },
     image: {
         width: windowWidth,
@@ -259,6 +292,13 @@
         backgroundColor: 'white',
         fontSize:30,
 
+    },
+    boxShadow: {
+        shadowColor: "#000",
+        shadowOffset: {width: -2, height: 4},
+        shadowOpacity: 0.2,
+        shadowRadius: 3,
+        elevation: 4,
     },
     loading: {
         position: 'absolute',
@@ -293,16 +333,6 @@
                     }}
                 />
             </TouchableOpacity>,
-        headerRight:
-            <TouchableOpacity
-                onPress={() => {params.downloadCurrentQuote(params.inViewPort)}}
-            >
-                <IconButton
-                    icon={require("@src/assets/img/download-large.png")}
-                    tintColor={"#4942e1"}
-                    style={{height: 25, marginRight:0}}
-                />
-            </TouchableOpacity>
     })
 }
 export default connect(mapStateToProps)(QuotesScreen);
Index: index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/index.js b/index.js
--- a/index.js	
+++ b/index.js	(date 1667616040832)
@@ -794,7 +794,7 @@
                     let tmpAchievements;
                     let acpTempIndex;
 
-                    updateDaily = today !== new moment.unix(state.progressReducer.lastPractice).format('YYYY-MM-DD');
+                    updateDaily = today !== state.progressReducer.lastPractice;
                     if(updateDaily)
                     {
                         console.log('21')
@@ -1058,6 +1058,18 @@
                                 }).sort((a, b) => {
                                     return a.claim_date > b.claim_date
                                 }),
+                            weekly: {
+                                days: acpTempAchievementState.weekly.days,
+                                list: acpTempAchievementState.weekly.list.sort((a, b) => {
+                                    return a.complete_date > b.complete_date
+                                }),
+                            },
+                            monthly: {
+                                days: acpTempAchievementState.monthly.days,
+                                list: acpTempAchievementState.monthly.list.sort((a, b) => {
+                                    return a.complete_date > b.complete_date
+                                }),
+                            }
                         },
                         progressReducer: acpTempProgressState,
                         practiceReducer: acpTempPracticeState
@@ -1163,7 +1175,11 @@
                     acwTempProgressState.latestUpdate = Math.floor(new Date().getTime() / 1000)
                     return {
                         ...state,
-                        achievementReducer: acwTempAchievementState,
+                        achievementReducer: {
+                            ...state.achievementReducer,
+                            weekly: acwTempAchievementState.weekly,
+                            monthly: acwTempAchievementState.monthly
+                        },
                         progressReducer: acwTempProgressState
                     };
                 case "ONENERGY_LESSON_COMPLETED":
Index: Components/Milestones.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Components/Milestones.js b/Components/Milestones.js
--- a/Components/Milestones.js	
+++ b/Components/Milestones.js	(date 1667599764507)
@@ -1,5 +1,4 @@
 import React from 'react';
-import {getApi} from "@src/services";
 import {connect, useSelector, useDispatch} from "react-redux";
 import {
     View,
@@ -33,24 +32,36 @@
     const dispatch = useDispatch();
 
     const playPause = () => {
-        let ding = new Sound('bonus_point.mp3', Sound.MAIN_BUNDLE,error => {
-            if (error) {
-                console.log('failed to load the sound', error);
-                return;
-            }
-            ding.play(success => {
-                ding.release();
-            });
-        });
+        if(Platform.OS==="android") {
+            let ding = new Sound('bonus_bell.mp3', Sound.MAIN_BUNDLE, error => {
+                if (error) {
+                    console.log('failed to load the sound', error);
+                    return;
+                }
+                ding.play(success => {
+                    ding.release();
+                });
+            });
+        }else if(Platform.OS === "ios"){
+            let ding = new Sound('https://media.onenergy.institute/audios/bonus_bell.mp3', null,error => {
+                if (error) {
+                    console.log('failed to load the sound', error);
+                    return;
+                }
+                ding.play(() => {
+                    ding.release();
+                });
+            });
+        }
     };
     const handleOnPress = (item, mode) => {
-        playPause();
-        if (Platform.OS !== "android") {
-            LayoutAnimation.configureNext(
-                LayoutAnimation.Presets.spring
-            );
-        }
-        if(item.complete_date&&!item.claim_date) {
+        if(item.complete_date&&!item.claim_date) {
+            playPause();
+            if (Platform.OS !== "android") {
+                LayoutAnimation.configureNext(
+                    LayoutAnimation.Presets.spring
+                );
+            }
             dispatch({
                 type: "ONENERGY_ACHIEVEMENT_CLAIM",
                 payload: {
Index: Components/QuestsDaily.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Components/QuestsDaily.js b/Components/QuestsDaily.js
--- a/Components/QuestsDaily.js	
+++ b/Components/QuestsDaily.js	(date 1667599764500)
@@ -31,26 +31,38 @@
     const achievementReducer = useSelector((state) => state.onenergyReducer?state.onenergyReducer.achievementReducer.achievements.filter(achievement => achievement.type === 'daily'):null);
     const dispatch = useDispatch();
     const playPause = () => {
-        let ding = new Sound('bonus_point.mp3', Sound.MAIN_BUNDLE,error => {
-            if (error) {
-                console.log('failed to load the sound', error);
-                return;
-            }
-            ding.play(success => {
-                ding.release();
-            });
-        });
+        if(Platform.OS==="android") {
+            let ding = new Sound('bonus_bell.mp3', Sound.MAIN_BUNDLE, error => {
+                if (error) {
+                    console.log('failed to load the sound', error);
+                    return;
+                }
+                ding.play(success => {
+                    ding.release();
+                });
+            });
+        }else if(Platform.OS === "ios"){
+            let ding = new Sound('https://media.onenergy.institute/audios/bonus_bell.mp3', null,error => {
+                if (error) {
+                    console.log('failed to load the sound', error);
+                    return;
+                }
+                ding.play(() => {
+                    ding.release();
+                });
+            });
+        }
     };
     const handleOnPress = (item, date, mode) => {
-        playPause();
-        if (Platform.OS !== "android") {
-            LayoutAnimation.configureNext(
-                LayoutAnimation.Presets.spring
-            );
-        }
         switch (mode) {
             case 'past':
                 if(item.complete_date!=='') {
+                    if (Platform.OS !== "android") {
+                        LayoutAnimation.configureNext(
+                            LayoutAnimation.Presets.spring
+                        );
+                    }
+                    playPause();
                     dispatch({
                         type: "ONENERGY_ACHIEVEMENT_CLAIM_DAILY",
                         payload: {
@@ -63,6 +75,12 @@
                 break;
             default:
                 if (item.complete_date !== '' && item.claim_date === '') {
+                    if (Platform.OS !== "android") {
+                        LayoutAnimation.configureNext(
+                            LayoutAnimation.Presets.spring
+                        );
+                    }
+                    playPause();
                     dispatch({
                         type: "ONENERGY_ACHIEVEMENT_CLAIM",
                         payload: {
Index: Components/QuestsWeekly.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Components/QuestsWeekly.js b/Components/QuestsWeekly.js
--- a/Components/QuestsWeekly.js	
+++ b/Components/QuestsWeekly.js	(date 1667616260852)
@@ -5,7 +5,7 @@
     Text,
     StyleSheet,
     SafeAreaView,
-    Image, ScrollView, TouchableWithoutFeedback
+    Image, ScrollView, TouchableWithoutFeedback, Platform
 } from 'react-native';
 import {scale} from "../Utils/scale";
 import {windowWidth} from "../Utils/Dimensions";
@@ -18,15 +18,27 @@
     const today = new moment().format('YYYY-MM-DD');
     const dispatch = useDispatch();
     const playPause = () => {
-        let ding = new Sound('bonus_claim.mp3', Sound.MAIN_BUNDLE,error => {
-            if (error) {
-                console.log('failed to load the sound', error);
-                return;
-            }
-            ding.play(success => {
-                ding.release();
-            });
-        });
+        if(Platform.OS==="android") {
+            let ding = new Sound('bonus_bell.mp3', Sound.MAIN_BUNDLE, error => {
+                if (error) {
+                    console.log('failed to load the sound', error);
+                    return;
+                }
+                ding.play(success => {
+                    ding.release();
+                });
+            });
+        }else if(Platform.OS === "ios"){
+            let ding = new Sound('https://media.onenergy.institute/audios/bonus_bell.mp3', null,error => {
+                if (error) {
+                    console.log('failed to load the sound', error);
+                    return;
+                }
+                ding.play(() => {
+                    ding.release();
+                });
+            });
+        }
     };
     return(
         <SafeAreaView style={styles.container}>
@@ -34,8 +46,8 @@
                 <Text style={styles.titleText}>Practice consecutively for 7 days to unlock this reward. Miss one day will reset the progress.</Text>
                 {Array(7).fill().map((_, idx) => 1 + idx).map((day,index)=>{
                     return (
-                        <View style={[styles.row, styles.boxShadow, {backgroundColor: achievementReducer?achievementReducer.days&&achievementReducer.days.length?achievementReducer.days[index]!==undefined&&achievementReducer.days[index]!==null&&achievementReducer.days[index]!==''?'#7de7fa':'#e6e6e8':'#e6e6e8':'#e6e6e8'}]} >
-                            <Text style={[styles.title,{textShadowColor: 'grey', textShadowRadius: 1, textShadowOffset: {width: -1,height: 1}, color:index===6?"gold":achievementReducer?achievementReducer.days&&achievementReducer.days.length?achievementReducer.days[index]!==undefined&&achievementReducer.days[index]!==null&&achievementReducer.days[index]!==''?'white':'black':'black':'black'}]}>Day {day} {index===6?'REWARD +20 Qi':''}</Text>
+                        <View style={[styles.row, styles.boxShadow, {backgroundColor: achievementReducer?achievementReducer.days&&achievementReducer.days.length?achievementReducer.days[index]!==undefined&&achievementReducer.days[index]!==null&&achievementReducer.days[index]!==''?'#8c78ff':'#e6e6e8':'#e6e6e8':'#e6e6e8'}]} >
+                            <Text style={[styles.title,{textShadowColor: 'grey', textShadowRadius: 1, textShadowOffset: {width: -1,height: 1}, color:index===6?"gold":achievementReducer?achievementReducer.days&&achievementReducer.days.length?achievementReducer.days[index]!==undefined&&achievementReducer.days[index]!==null&&achievementReducer.days[index]!==''?'white':'white':'white':'white'}]}>Day {day} {index===6?'REWARD +20 Qi':''}</Text>
                             <View style={{flexDirection:"row", justifyContent:"center", alignItems:"center"}}>
                                 <Text style={{marginRight:10, color: '#FFF', textShadowColor: 'grey', textShadowRadius: 1, textShadowOffset: {width: -1,height: 1}}}>{achievementReducer?achievementReducer.days&&achievementReducer.days.length?achievementReducer.days[index]!==undefined&&achievementReducer.days[index]!==null&&achievementReducer.days[index]!==''?achievementReducer.days[index]:'':'':''}</Text>
                                 {
Index: Components/QiPointHeader.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Components/QiPointHeader.js b/Components/QiPointHeader.js
--- a/Components/QiPointHeader.js	
+++ b/Components/QiPointHeader.js	(date 1667601820362)
@@ -6,11 +6,10 @@
 import {scale} from '../Utils/scale';
 
 const QiPointHeader = (props) => {
-    const progressReducer = useSelector((state) => state.onenergyReducer?state.onenergyReducer.progressReducer.points.qi:0);
-    const optionData = useSelector((state) => state.settings.settings.onenergy_option);
+    const progressReducer = useSelector((state) => state.onenergyReducer&&state.onenergyReducer.progressReducer.points.qi?state.onenergyReducer.progressReducer.points.qi:0);
     return (
         <View style={{flexDirection:"row", justifyContent:"flex-end", alignItems:"center", marginRight:scale(15)}}>
-            <FastImage source={{uri:'https://assets.onenergy.institute/wp-content/uploads/2020/07/gamipress-icon-ray-material-54x54.png'}} style={{width:16, height:16}} />
+            <FastImage source={require('../assets/images/icon-ray.png')} style={{width:16, height:16}} />
             <AnimatedNumber style={{color:"black"}} value={progressReducer} />
         </View>
     );
Index: Components/QuestsMonthly.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Components/QuestsMonthly.js b/Components/QuestsMonthly.js
--- a/Components/QuestsMonthly.js	
+++ b/Components/QuestsMonthly.js	(date 1667616260850)
@@ -5,7 +5,7 @@
     Text,
     StyleSheet,
     SafeAreaView,
-    Image, ScrollView, TouchableWithoutFeedback
+    Image, ScrollView, Platform,TouchableWithoutFeedback
 } from 'react-native';
 import {scale} from "../Utils/scale";
 import {windowWidth} from "../Utils/Dimensions";
@@ -18,15 +18,27 @@
     const today = new moment().format('YYYY-MM-DD');
     const dispatch = useDispatch();
     const playPause = () => {
-        let ding = new Sound('bonus_point.mp3', Sound.MAIN_BUNDLE,error => {
-            if (error) {
-                console.log('failed to load the sound', error);
-                return;
-            }
-            ding.play(success => {
-                ding.release();
-            });
-        });
+        if(Platform.OS==="android") {
+            let ding = new Sound('bonus_bell.mp3', Sound.MAIN_BUNDLE, error => {
+                if (error) {
+                    console.log('failed to load the sound', error);
+                    return;
+                }
+                ding.play(success => {
+                    ding.release();
+                });
+            });
+        }else if(Platform.OS === "ios"){
+            let ding = new Sound('https://media.onenergy.institute/audios/bonus_bell.mp3', null,error => {
+                if (error) {
+                    console.log('failed to load the sound', error);
+                    return;
+                }
+                ding.play(() => {
+                    ding.release();
+                });
+            });
+        }
     };
     return(
         <SafeAreaView style={styles.container}>
@@ -35,7 +47,7 @@
                 <View style={styles.daysContainer}>
                 {Array(30).fill().map((_, idx) => 1 + idx).map((day,index)=>{
                     return (
-                        <View style={[styles.row, styles.boxShadow, {backgroundColor: achievementReducer?achievementReducer.days&&achievementReducer.days.length?achievementReducer.days[index]!==undefined&&achievementReducer.days[index]!==null&&achievementReducer.days[index]!==''?'#7de7fa':'#e6e6e8':'#e6e6e8':'#e6e6e8'}]} >
+                        <View style={[styles.row, styles.boxShadow, {backgroundColor: achievementReducer?achievementReducer.days&&achievementReducer.days.length?achievementReducer.days[index]!==undefined&&achievementReducer.days[index]!==null&&achievementReducer.days[index]!==''?'#8c78ff':'#e6e6e8':'#e6e6e8':'#e6e6e8'}]} >
                             <View style={{flexDirection:"row", justifyContent:"center", alignItems:"center"}}>
                                 {
 
@@ -49,7 +61,7 @@
                                     <Image source={require("@src/assets/img/radio_unchecked_icon.png")}/>
                                 }
                             </View>
-                            <Text style={{textShadowColor: 'grey', textShadowRadius: 1, textShadowOffset: {width: -1,height: 1}, color:index===30?"gold":achievementReducer?achievementReducer.days&&achievementReducer.days.length?achievementReducer.days[index]!==undefined&&achievementReducer.days[index]!==null&&achievementReducer.days[index]!==''?'white':'black':'black':'black'}}>Day {day}</Text>
+                            <Text style={{textShadowColor: 'grey', textShadowRadius: 1, textShadowOffset: {width: -1,height: 1}, color:index===30?"gold":achievementReducer?achievementReducer.days&&achievementReducer.days.length?achievementReducer.days[index]!==undefined&&achievementReducer.days[index]!==null&&achievementReducer.days[index]!==''?'white':'white':'white':'white'}}>Day {day}</Text>
                         </View>
                     )
                 })}
Index: Components/AchievementItem.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Components/AchievementItem.js b/Components/AchievementItem.js
--- a/Components/AchievementItem.js	
+++ b/Components/AchievementItem.js	(date 1667616260842)
@@ -4,7 +4,7 @@
     View,
     Text,
     StyleSheet,
-    TouchableOpacity,
+    TouchableWithoutFeedback,
 } from 'react-native';
 import {scale, windowWidth} from "../Utils/scale";
 import * as Progress from 'react-native-progress';
@@ -28,7 +28,7 @@
                     :null}
                 {mode!=='past'&&item.claim_date===''?
                     <View style={{marginTop: 10, opacity: item.claim_date!==''?0:100  }}>
-                        <Progress.Bar showsText={true} borderColor={"#4942e1"} color={item.complete_date===today?"lightgreen":"#7de7fa"} unfilledColor={"black"} borderRadius={9}
+                        <Progress.Bar showsText={true} borderColor={"#4942e1"} color={item.complete_date===today?"lightgreen":"#8c78ff"} unfilledColor={"black"} borderRadius={9}
                                       progress={item.complete_date===today?item.step:0 / item.total}
                                       width={windowWidth/2} height={scale(16)}/>
                         <View
@@ -39,12 +39,12 @@
                     </View>
                     :null}
             </View>
-            <TouchableOpacity
+            <TouchableWithoutFeedback
                 onPress={() => {
                     handleOnPress(item, date, mode);
                 }}
             >
-                <View style={[styles.rowRight, {backgroundColor:item.complete_date===''?'#7de7fa':mode==='past'||item.claim_date===''?'gold':'gray'}]}>
+                <View style={[styles.rowRight, {backgroundColor:item.complete_date===''?'#8c78ff':mode==='past'||item.claim_date===''?'gold':'gray'}]}>
                     {mode!=='past'&&item.claim_date!==''?
                         <>
                             <Text
@@ -111,7 +111,7 @@
                         </>
                         :null}
                 </View>
-            </TouchableOpacity>
+            </TouchableWithoutFeedback>
         </View>
     );
 };
@@ -136,7 +136,7 @@
         justifyContent: 'center',
         width: (windowWidth - scale(30))/3,
         height: scale(70),
-        backgroundColor: '#7de7fa',
+        backgroundColor: '#8c78ff',
     },
     rowLeft: {
         marginVertical: 0,
